component-runtime-manager/src/test/java/org/talend/sdk/component/runtime/manager/service/LocalCacheServiceTest.java

```diff
@@ -15,46 +15,167 @@
  */
 package org.talend.sdk.component.runtime.manager.service;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.fail;
-import static org.talend.sdk.component.runtime.manager.test.Serializer.roundTrip;
-
 import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Supplier;
 
+import org.apache.xbean.propertyeditor.PropertyEditorRegistry;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.talend.sdk.component.api.service.cache.LocalCache;
+import org.talend.sdk.component.api.service.cache.LocalCache.Element;
+import org.talend.sdk.component.api.service.configuration.LocalConfiguration;
+import org.talend.sdk.component.api.service.injector.Injector;
+import org.talend.sdk.component.runtime.manager.asm.ProxyGenerator;
+import org.talend.sdk.component.runtime.manager.reflect.ParameterModelService;
+import org.talend.sdk.component.runtime.manager.reflect.ReflectionService;
 import org.talend.sdk.component.runtime.manager.serialization.DynamicContainerFinder;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.talend.sdk.component.runtime.manager.test.Serializer.roundTrip;
+
 class LocalCacheServiceTest {
 
+
+    private boolean active;
+
+    private int  intervale;
+
+    private LocalConfiguration cacheConfig = new LocalConfiguration() {
+        @Override
+        public String get(String key) {
+            if ("test.talend.component.manager.services.cache.eviction.defaultEvictionInterval".equals(key)) {
+                return String.valueOf(intervale);
+            }
+            if ("test.talend.component.manager.services.cache.eviction.active".equals(key)) {
+                return String.valueOf(active);
+            }
+            throw new IllegalArgumentException("unknown key '" + key + "'");
+        }
+
+        @Override
+        public Set<String> keys() {
```

Comment:
not needed, note that you can likely reuse org.talend.sdk.component.runtime.manager.service.configuration.PropertiesConfiguration and handle the configuration in a Properties instead instead

Reply:
keeping LocalConfiguration like this allow me to have dynamic value (active, interval) for unit test

Reply:
same using the properties, you just do "setProperty" instead of "=" and it reduces the boilerplate significatively IMHO

Reply:
removed